service/policy/objects.proto

```diff
@@ -265,16 +265,36 @@ message KeyAccessServer {
   common.Metadata metadata = 100;
 }
 
+// A KAS public key and some associated metadata for further identifcation
+message KasPublicKey {
+  // x509 ASN.1 content in PEM envelope, usually
+  string pem = 1;
+
+  // A unique string identifier for this key
+  string kid = 2;
+
+  // A known algorithm type with any additional parameters encoded
```

Comment:
can we specify what kind of encoding any additional parameters should follow?

Reply:
Currently, this is functionally an enum with the values `ec:secp256r1` or `rsa:2048` - in the case EC + curve and bits, RSA with bits. I'll call those out explicitly

Reply:
If that's the case should we use enums in the proto

Reply:
please no

Reply:
We are an open product and this is intended to be extensible and useful. by adding explicit values here we prevent ourselves from adding new types that are experimental or unsupported by opentdf

Reply:
Let's see how it works out.

Reply:
Eh it isn't that big of a change so I'll go ahead with this. In the end I do still want to replace this whole thing with a JWK Set, which will be definitely be safest to use strings. We don't want policy service to have too big of an opinion about these things because it will cause surprising breakages when round tripping through various non-compliant protocol buffer implementations.

https://github.com/opentdf/platform/pull/1346/commits/0a902c605a802a7fe6136212b130c1994d3c9530